File: sdkwork-im-sdk-flutter/composed/lib/src/presence_module.dart
import 'package:im_sdk_generated/im_sdk_generated.dart';

import 'context.dart';

class ImPresenceModule {
  final ImSdkContext context;

  ImPresenceModule(this.context);

  Future<PresenceSnapshotView?> heartbeat(PresenceDeviceRequest body) {
    return context.transportClient.presence.heartbeat(body);
  }

  Future<PresenceSnapshotView?> current() {
    return context.transportClient.presence.getPresenceMe();
  }
}

File: sdkwork-im-sdk-rust/composed/src/presence_module.rs
use crate::ImSdkContext;
use sdkwork_im_sdk_generated::{PresenceDeviceRequest, PresenceSnapshotView, SdkworkError};

#[derive(Clone)]
pub struct ImPresenceModule {
  context: ImSdkContext,
}

impl ImPresenceModule {
  pub(crate) fn new(context: ImSdkContext) -> Self {
    Self { context }
  }

  pub async fn heartbeat(
    &self,
    body: PresenceDeviceRequest,
  ) -> Result<PresenceSnapshotView, SdkworkError> {
    self.context.transport_client().presence().heartbeat(&body).await
  }

  pub async fn me(&self) -> Result<PresenceSnapshotView, SdkworkError> {
    self.context.transport_client().presence().get_presence_me().await
  }
}

